application/controllers/artifacts_controller.py

```diff
@@ -38,30 +41,31 @@ def show(self, object_id):
 
     @use_args(artifacts_validator.search_args())
     def index(self, params):
-        "Logic for querying several artifacts"
+        """Logic for querying several artifacts"""
 
         artifacts = Artifact.search(params)
 
         return {"images": respond_with(artifacts)}, 200
 
     @use_args(artifacts_validator.create_args())
     def create(self, params):
-        "Logic for creating an artifact"
+        """Logic for creating an artifact"""
         params["file_date"] = datetime.datetime.now()
         uploaded_file = params["file"]
         filename = str(uuid.uuid4()) + "_" + \
-            werkzeug.utils.secure_filename(uploaded_file.filename)
+                   werkzeug.utils.secure_filename(uploaded_file.filename)
         file_url = os.path.join(current_app.config["UPLOAD_FOLDER"], filename)
         uploaded_file.save(file_url)
         params["file_url"] = filename
         artifact = Artifact(params)
 
         artifact.save()
+        recognize_image(file_url, artifact)
```

Comment:
This method is getting pretty long. Tbh it was way too long before already.  
Simply doing some method extraction would help imo. So something like this?
```suggestion
        params["file_url"] = upload_file()
        artifact = Artifact(params)
        artifact.save()
        recognize_image(params["file_url", artifact)
```

also consider renaming `recognize_image()` to something like 'add_google_vision_tags()'

Reply:
I've not worked with the artifacts_controller up until now so I will look into where to put the upload_file function, and what has to go into it. I think extracting that as a method sounds reasonable. 
On the rename: 
I would like to keep the rest of ejija agnostic as to what type of recognition is actually used.
This allows us to only change what (private) function the worker thread in recognize_image calls between the local and the google api image recognition. 
This will enable A/B testing between the 2 versions and using them in parrallel, by only adjusting one line in the recognize_image function.

Reply:
If you don't want to specify google for that reason, that's okay but on first sight `recognize_image` doesn't really tell me what it does. For me that sounds more like, the function will check weather the system already knows the image or something like that and not automatically add tags. So if you want to rename to `auto_add_tags` that's fine with me as well.

Reply:
Sounds fine.